services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/FileBasedTenantService.java

```diff
@@ -306,6 +306,10 @@ public void add(final String tenantId, final JsonObject tenantSpec, final Handle
 
         if (tenants.containsKey(tenantId)) {
             return TenantResult.from(HttpURLConnection.HTTP_CONFLICT);
+        } else if (!isValidRequestPayload(tenantSpec)) {
```

Comment:
this should already have been verified in `CompleteBaseTenantService.processAddRequest()`. In fact, not having to validate in the concrete implementation is the reason why you want to add the check to the super class in the first place, isn't it?

Reply:
exactly! I added it here because the `FileBaseTenantServiceTest` directly calls the `add` method of the FileBaseTenantService without going through the `CompleteBaseTenantService.processAddRequest()`

Reply:
can we assume that all tenant API requests (add, update, delete, etc) must come via the Vertx event bus?

Reply:
i see.. please ignore :-)